auth/src/main/java/org/aerogear/auth/credentials/OIDCCredentials.java

```diff
@@ -32,6 +47,69 @@ public String getRefreshToken() {
         return authState.getRefreshToken();
     }
 
+    public IIntegrityCheckParameters getIntegrityCheckParameters() { return this.integrityCheckParameters; }
+
+    /**
+     * Verify the authenticity of a JWT token against integrity parameters (provided config).
+     * @param jwtToken The JWT token to verify.
+     * @return <code>true</code> if the token integrity is good.
+     */
+    public boolean verifyToken(final String jwtToken) {
+        String issuer = integrityCheckParameters.getIssuer();
+        String audience = integrityCheckParameters.getAudience();
+        String publicKey = integrityCheckParameters.getPublicKey();
+        return verifyToken(jwtToken, publicKey, issuer, audience);
+    }
+
+    /**
+     * A function to verify the authenticity of a JWT token against a public key, expected issuer and audience.
+     * @param jwtToken - The JWT Token to Verify.
+     * @param publicKey - The Public Key from Keycloak, without the Begin/End tags.
+     * @param issuer - The expected Issuer of the JWT
+     * @param audience - The expected Audience of the JWT
+     * @return <code>true</code> if the token integrity is good.
+     */
+    public boolean verifyToken(final String jwtToken, final String publicKey, final String issuer, final String audience) {
+        // add the Begin/End tags to the public key generated from Keycloak
+        String beginPublicKey = "-----BEGIN PUBLIC KEY-----";
+        String endPublicKey = "-----END PUBLIC KEY-----";
```

Comment:
these should be private static final constant fields instead of local variables.